core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/workspacemanager/HttpWorkspaceManagerDAOSpec.scala

```diff
@@ -319,4 +319,59 @@ class HttpWorkspaceManagerDAOSpec
                                                     ArgumentMatchers.eq("test_job_id")
     )
   }
+
+  behavior of "listWorkspaces"
+
+  it should "should only make a single request when no workspaces are returned" in {
+    val workspaceApi = mock[WorkspaceApi]
+    val responseList = new WorkspaceDescriptionList().workspaces(List().asJava)
+    when(workspaceApi.listWorkspaces(ArgumentMatchers.eq(0), anyInt(), any())).thenReturn(responseList)
+    val wsmDao = new HttpWorkspaceManagerDAO(getApiClientProvider(workspaceApi = workspaceApi))
+    wsmDao.listWorkspaces(testContext) shouldBe empty
```

Comment:
Readability nits: consider splitting these into the usual arrange/act/assert structure rather than a single block (I don't feel a compelling need for the terra-ui `// arrange //act //assert` style comments though).